man/man8/zdb.8

```diff
@@ -97,6 +97,11 @@ characters, it is interpreted as a pool name.
 The root dataset can be specified as
 .Ar pool Ns /
 .Pq pool name followed by a slash .
+As a special case, the 
+.Ar dataset 
+argument may contain the keyword
+.Qq Sy dataset
+followed by the numeric id of the dataset.
```

Comment:
Do we really want to add the dataset keyword?  Since pool names cannot begin with a leading digit we can determine when an objset-id or dataset ame was passed and do the right thing.  This would be a little more intuitive since the existing syntax would work when an objset-id was substituted.

Reply:
I was open to this, but started playing around with the command line, and there's going to be some ambiguity i.e.
{code}
# zdb tank -d 236
Dataset mos [META], ID 0, cr_txg 4, 6.04M, 619 objects

    Object  lvl   iblk   dblk  dsize  dnsize  lsize   %full  type
       236    1   128K    512      0     512    512  100.00  zap
{\code}
So I think a keyword is needed to express the intent of looking up a dataset by id.

Reply:
I was thinking something like this _should_ be workable and consistent with the existing usage. 

```
Usage:	zdb [-AbcdDFGhikLMPsvX] [-e [-V] [-p <path> ...]] [-I <inflight I/Os>]
		[-o <var>=<value>]... [-t <txg>] [-U <cache>] [-x <dumpdir>]
		[<pool>[/<dataset | objset id>] [<object> ...]]

```

Then assuming the dataset `fish` uses the objset id `0x20 / 32d`, and your interested in object 236 any of the following would work (and not be ambiguous).  Or are the other cases where this wouldn't work?

```
$ zdb -d tank/fish 236
$ zdb -d tank/0x20 236
$ zdb -d tank/32 236
```

Reply:
Having some trouble getting zdb tank/0 <object>.  We already have this:
```
# zdb -dd tank 2
Dataset mos [META], ID 0, cr_txg 4, 6.13M, 617 objects

    Object  lvl   iblk   dblk  dsize  dnsize  lsize   %full  type
         2    1   128K   128K     3K     512   128K  100.00  bpobj
```
So maybe we don't need to support tank/0 2

Reply:
Ok I got it working so the user can do either way.

Reply:
Great.  If it's particularly problematic I don't think it's a must have, but it would be nice to support.